src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/LocationResource.kt

```diff
@@ -58,8 +60,43 @@ class LocationResource : EventBaseResource() {
     @Schema(description = "The location Id", example = "de91dfa7-821f-4552-a427-bf2f32eafeb0", required = true)
     @PathVariable
     id: UUID,
-  ): LocationDetail {
-    return auditWrapper(AuditType.LOCATION_RETRIEVED, id.toString()) { LocationDetail(id, "A-1-001") }
+  ): LocationDTO {
+    return auditWrapper(AuditType.LOCATION_RETRIEVED, id.toString()) {
+      locationService.getLocationById(id) ?: throw LocationNotFoundException(id.toString())
+    }
+  }
+
+  @GetMapping("/key/{key}")
+  @ResponseStatus(HttpStatus.OK)
+  @PreAuthorize("hasRole('ROLE_VIEW_LOCATIONS')")
+  @Operation(
+    summary = "Returns location information for this key",
+    description = "Requires role VIEW_LOCATIONS",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Returns location",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Data not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getLocationByKey(
+    @Schema(description = "Location Key", example = "MDI-A-1-001", required = true)
```

Comment:
Do we need validation on the format of this variable to ensure the `.split` returns a value for `code`?

Reply:
Yes I think we will - I add some simple validation in next pr